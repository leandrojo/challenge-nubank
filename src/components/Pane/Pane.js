import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const StyledPane = styled.div`
  ${p => p.theme.components.pane}

  display: flex;
  flex-direction: column;
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  width: 100%;

  ${({ elevation }) => {
    switch (elevation) {
      case 2:
        return css`
          box-shadow:
            0 3px 6px rgba(0,0,0,0.16),
            0 3px 6px rgba(0,0,0,0.23);
        `;
      case 3:
        return css`
          box-shadow:
            0 10px 20px rgba(0,0,0,0.19),
            0 6px 6px rgba(0,0,0,0.23);
        `;
      case 4:
        return css`
          box-shadow:
            0 14px 28px rgba(0,0,0,0.25),
            0 10px 10px rgba(0,0,0,0.22);
        `;
      case 5:
        return css`
          box-shadow:
            0 19px 38px rgba(0,0,0,0.30),
            0 15px 12px rgba(0,0,0,0.22);
        `;
      case 1:
      default: {
        return css`
          box-shadow:
            0 1px 3px rgba(0,0,0,0.12),
            0 1px 2px rgba(0,0,0,0.24);
        `;
      }
    }
  }}
`;

const StyledHeader = styled.header`
  display: flex;
  flex: 1;
  justify-content: center;
  
`;

const StyledContent = styled.div`
  padding: 30px;
`;

const Pane = ({ children, elevation, ...props }) => (
  <StyledPane {...props} elevation={elevation}>
    {children}
  </StyledPane>
);

Pane.Content = StyledContent;
Pane.Header = StyledHeader;

Pane.defaultProps = {
  children: null,
  elevation: 1,
};

Pane.propTypes = {
  children: PropTypes.node,
  elevation: PropTypes.number,
};

export default Pane;
