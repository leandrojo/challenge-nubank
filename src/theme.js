const fontFamily = 'Rubik, sans-serif';

const theme = {
  backgroundColor: '#f5f5f5',
  colors: {
    primary: '#7f39b7',
    grayExtraDark: '#555',
    grayDark: '#767676',
    gray: '#999',
    grayLight: '#ccc',
    grayExtraLight: '#FAFAFA',
  },
  fontFamily,
  fontSize: '1em',
  typography: {
    heading: {
      color: '#252226',
    },
    paragraph: {
      color: '#716d73',
    },
  },
};

theme.components = {
  input: {
    css: {
      backgroundColor: 'transparent',
      fontFamily,
      fontSize: '1.4em',
      fontWeight: 300,
    },
  },
  pane: {
    backgroundColor: '#fff',
  },
  range: {
    emptyColor: theme.colors.grayLight,
  },
};

export default theme;
