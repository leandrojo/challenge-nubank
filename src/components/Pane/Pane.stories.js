/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';

import { Box } from '..';
import Pane from './Pane';

export default {
  title: 'components/Pane',
};

export const elevations = () => (
  <>
    {[1, 2, 3, 4, 5].map(value => (
      <Box css={{ margin: '20px', padding: '20px' }}>
        <Pane elevation={value}>
          <Box css={{ alignItems: 'center', justifyContent: 'center', padding: '10px' }}>
            <span>{`Elevation ${value}`}</span>
          </Box>
        </Pane>
      </Box>
    ))}
  </>
);

elevations.story = {
  name: 'elevations',
};
