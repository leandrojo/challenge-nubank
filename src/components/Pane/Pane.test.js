/* eslint-env jest */

import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import theme from 'theme';

import Pane from './Pane';

describe('Pane', () => {
  it('with text', () => {
    const Component = () => (
      <ThemeProvider theme={theme}>
        <Pane>Test Message</Pane>
      </ThemeProvider>
    );

    const expected = renderer.create(<Component />).toJSON();

    expect(expected).toMatchSnapshot();
  });
});
