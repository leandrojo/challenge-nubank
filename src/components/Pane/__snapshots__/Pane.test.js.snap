// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pane with text 1`] = `
<div
  className="sc-bdVaJa iBVmJm"
  elevation={1}
>
  Test Message
</div>
`;
